documentation/examples/remote_storage/remote_storage_adapter/main.go

```diff
@@ -217,20 +217,8 @@ func serve(addr string, writers []writer, readers []reader) error {
 	})
 
 	http.HandleFunc("/read", func(w http.ResponseWriter, r *http.Request) {
-		compressed, err := ioutil.ReadAll(r.Body)
+		req, err := remote.DecodeRequest(r)
```

Comment:
I can move this to a separate package, its quite decoupled.  How about "codec" package under remote?